tgui/packages/tgui/interfaces/MessageMonitor.tsx

```diff
@@ -0,0 +1,385 @@
+                <Table.Cell>{request.id_auth}</Table.Cell>
+              </Table.Row>
+            ))}
+          </Table>
+        </Section>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const MessageLogsScreen = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { messages = [] } = data;
+  return (
+    <Stack fill vertical>
+      <Stack.Item grow>
+        <Section
+          fill
+          scrollable
+          title="Stored Messages"
+          buttons={
+            <Button
+              content="Main Menu"
+              icon="home"
+              onClick={() => act('return_home')}
+            />
+          }>
+          <Table>
+            <Table.Row header>
+              <Table.Cell>Delete</Table.Cell>
+              <Table.Cell>Sender</Table.Cell>
+              <Table.Cell>Recipient</Table.Cell>
+              <Table.Cell>Message</Table.Cell>
+            </Table.Row>
+            {messages?.map((message) => (
+              <Table.Row key={message.ref} className="candystripe">
+                <Table.Cell>
+                  <Button
+                    icon="trash"
+                    color="red"
+                    onClick={() => act('delete_message', { ref: message.ref })}
+                  />
+                </Table.Cell>
+                <Table.Cell>{message.sender}</Table.Cell>
+                <Table.Cell>{message.recipient}</Table.Cell>
+                <Table.Cell>{message.message}</Table.Cell>
+              </Table.Row>
+            ))}
+          </Table>
+        </Section>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const HackedScreen = (props, context) => {
+  return (
+    <Stack.Item grow>
+      <Stack fill vertical>
+        <Stack.Item grow />
+        <Stack.Item align="center" grow>
+          <Box color="red" fontSize="18px" bold mt={5}>
+            E%*#OR: CRIT#%^CAL Ss&@STEM Fai++~|URE...
+          </Box>
+          <Box color="red" fontSize="18px" bold mt={5}>
+            EN*bLING u&*E REB-00T ProT/.\;L...
+          </Box>
+        </Stack.Item>
+      </Stack>
+    </Stack.Item>
+  );
+};
+
+const MainScreenAuth = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { status, is_malf, password } = data;
+  const [auth_password, setPassword] = useLocalState(
+    context,
+    'input_password',
+    password
+  );
+  return (
+    <>
+      <Stack.Item>
+        <Section>
+          <Input
+            value={auth_password}
+            onInput={(e, value) => setPassword(value)}
+            placeholder="Password"
+          />
+          <Button
+            content={'Logout'}
+            onClick={() => act('auth', { auth_password: auth_password })}
+          />
+          <Button
+            icon={status ? 'power-off' : 'times'}
+            content={status ? 'ON' : 'OFF'}
+            color={status ? 'green' : 'red'}
+            onClick={() => act('turn_server')}
+          />
+          {is_malf === 1 && (
```

Comment:
this is a booleanlike. while you're correct it is a 1 or 0, the way to check for this is `!!is_malf`